bin/bootstrap-cloud

```diff
@@ -192,9 +213,120 @@ if [[ ${__cloud} == aws ]]; then
 	-e "s#some-context#aws-${__name}#" \
         env.auto.tfvars.example > env.auto.tfvars
   fi
+  ;;
+  gcp)
+    check_gcp_prerequisites
+    
+    # Config file has to exist at this point
+    if [[ ! -e ${__root}/config.yaml ]]; then
+      echo "config.yaml not found, exiting..."; exit 1
+    fi
+    # Get the project ID
+    __project_id=$(grep ^terraform_project_id: ${__root}/config.yaml | awk '{print $2}')
+    if [[ -z ${__project_id} ]]; then
+      __project_id=$(gcloud config get-value project)
+      echo "terraform_project_id: ${__project_id}" >> ${__root}/config.yaml
+    fi
+    
+    # Get the billing account
+    __billing_account=$(grep ^billing_account: ${__root}/config.yaml | awk '{print $2}')
+    if [[ -z ${__billing_account} ]]; then
+      __billing_account=$(gcloud billing projects describe ${__project_id} --format="value(billingAccountName)" | awk -F'/' '{print $2}')
+      echo "billing_account: ${__billing_account}" >> ${__root}/config.yaml
+    fi
+    # Get the organization ID
+    __organization_id=$(grep ^organization_id: ${__root}/config.yaml | awk '{print $2}')
+    if [[ -z ${__organization_id} ]]; then
+      __organization_id=$(gcloud projects describe ${__project_id} --format="value(parent.id)")
+      echo "organization_id: ${__organization_id}" >> ${__root}/config.yaml
+    fi
+    
+    set -e
+    cd ${__root}/bootstrap/gcp-terraform
+    terraform init
+    terraform validate
+    set +e
+    terraform plan -detailed-exitcode -input=false  
+    
+    __ret=$?
+    case ${__ret} in
+      2) terraform apply -auto-approve ;;
+      0) echo terraform apply skipped with no changes ;;
+      *) echo terraform plan result ${__ret} && exit ${__ret} ;;
+    esac
+    __ret=$?
+    case ${__ret} in
+      0) echo terraform apply successful ;;
+      *) echo terraform apply result ${__ret} && exit ${__ret} ;;
+    esac
 
-else # k3d
+    set -e
+    __bucket=$(terraform output -raw terraform_state_bucket)
+    __kmskey=$(terraform output -raw terraform_state_crypto_key)
+    __project_id=$(terraform output -raw terraform_project_id)
+    if [[ ${__bucket} =~ bucket-tfstate ]]; then
+      echo Terraform state bucket: ${__bucket}
+    else
+      echo Could not detect state bucket && exit 1
+    fi
+    if [[ ${__kmskey} =~ terraform-state-bucket ]]; then
+      echo Terraform state KMS key: ${__kmskey}
+    else
+      echo Could not detect state KMS key && exit 1
+    fi
+    if [[ -n ${__project_id} ]]; then
```

Comment:
This will never be empty as you're reusing a variable which we set before running terraform.